docs/api/app.md

```diff
@@ -1,87 +1,98 @@
 # `app()`
 
-**_Definition:_**
-
-> A function that initializes and mounts a Hyperapp application.
-
-**_Import & Usage:_**
-
-```js
-import { app } from "hyperapp"
-
-// ...
-
-app(props)
-```
-
-**_Signature & Parameters:_**
+Initializes and mounts a Hyperapp application.
 
 ```elm
 app : ({ Init, View, Node, Subscriptions?, Dispatch? }) -> DispatchFn
 ```
 
-| Parameter       | Type   | Required? |
-| --------------- | ------ | --------- |
-| [props](#props) | Object | yes :100: |
+| Prop                            | Type                                                                      | Required? |
```

Comment:
Would there be any use case for an app with no view & node ?

Reply:
@zaceno @kofifus It turns out this is a little more complicated. I've gone through [all the possible JavaScript usage scenarios for `app()` here](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvAHiwwjQAIIATAXgA6IDAAdRwgHwsA9O06TBaSiDgxY1YhHoJEIAIyIADCAC+FdNlx78AKwRU6DJsTwQso2gCdiXYF0IKLjFRIMYAD19TLjAvWiwuYRJiUThEGRkAVzRRAGsAc3w6LBlCAE9RGC8Q4SUlJzhfTghffj8Y2lpERJBGRuEuU3qdXwA3CBgAdy42gApGjEYAShnJANnhCUo-cy4I4nniRZh8ME6lpeG0Rq40Wl4YGa5eWmpMnAZ8fJhiAFFYD7EABCZQAkrwNiJxMJLmgGr44JkAEZwaheCCiLQ6J6zAD6C2Wqy4yAAulcbrwIHBRItqIQcZTqbTCCt+GtGTTiHS6mgMlwAML3R7EQiLHhwLjFQEwXhcJHqWjTagYTJqLhebJaHBcKpxLxwfA8vkhWawk3AUxm8SzfzNKKw43W21oFpBcZTQZW0Q2ngu4hBO4PT1Kc2+11cREotEYrHXYOYJ1h-3PKmcunx0N2t0TSYBoXxx3e53h925iPI1HozHaOOWpSFn1ZriloIc5kZxNNwMwIKRysxmsSusJotJvMPVup9vDzN+3sV6PVnSTplc+nDhvF5Mt25C+dRquxocOmTBTtz5s58c9lOr9Mb0+zktX8sHgfL29p9dexsX7v7-sl2uFcvwLR9z2fKZrwAxcjxA6cTzPUdSzApCfR3btUNDHc+1gwcsMTHc2zXAjkJff9X0Ao9SPQ8i90-BD63AsioN3CdKLwj9iPvH9-EwmdEwo3DD3wgTRwo7jvxDQT6OE99gIYkixJ9OSgOPaTR1UuDFJ4jSfUk+MVDUDQjzwABWRAAGYADYzAsEBMBwPAijgBwaHoRhmD0MwSVMIA). The demo shows which usages for `app()` cause errors and which do not. Some of them are surprising. For example, the line that tries to use every property at the same time, `app({ init, view, node, subscriptions, dispatch })`, causes an error.

Reply:
@zaceno @kofifus @jorgebucaran Okay, apologies for my original demo which was constructed badly! Anyway, [here is an updated one which again shows all possible JavaScript usages for `app()`](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvAHiwwjQAIIATAXgA6IDAAdRARgAswgHwsA9O06zBaNh258hI8RIDscxcrSr1JngOFjRAJgAMRpZrMbOlnTdsSnJ1xe1rcVsAZl8XNTctK107GRB5ZxVIgJivADZw5PNNDyC7QwTjCJz3QNjbAE4s0xTc8psQ2xr-erTxELCipNrS6M8O+MS-OrL20RCAVhbR-vyQzO6RvrzYkMLhkqjVxoAOGbRKEDgYWGpiCHoERBAJdMR7EABfCnRsXBv8ACsEKjoGJjEPAQLCiWgAJ2IXGAXEIFC4NnhjAAHlCnlwwODaFguMISMRRHBEAoFABXNCiADWAHN8HQsApCABPUQwcE2YRqNT-OBQzgQKH8aEY2i0RC4kCMXnCLhPblXKEANwgMAA7lwhQAKXkYRgASg1slhmuEomE8OALy4KOI2uIupg+DAor1evlaF5XDgpIARnBqOCIKILlcNVxNQB9HX6w1cZAAXXdnt4EDgol11EIYc1KbTGcIBv4Rtz6eIma5aBJXAAwrReDBrYRdTw4Fx6TgAbwuD7TrR1dQMKSTlxweSLjguGyseC4PgK1WbJq3ZhxJrLcvFzD+WjlwvV1u0AL4cq1bKN-ueIfiPC0HWYOLeLRqKSO8R8NSYMQAKKwV8AISZABJXgTViaRhANJ5z1ENdLyPL1fX9QNg0uD0zzUTc4OvLgS3zdCVxgg94JPVUbzvB8nxfQF30-H8YH-ICQPyAwIPwvdCKw48VVIhC-QDIMQzQqC1HY2Dty4tV4Vwsss2EgixKvMj6wo59Xxo79f0BADgNArxHBAPV4W9PjkME1s5Mw8SuFvZScMotSPw0+itMY3Tgh8AypNTUtM3wyzFN4pCBNQuAvLzGS-IvKySKU+87NU6jHLohidPyUIIKMxD+JQ0MLKigKYus8j4qohh1OSlzUoqeJDJw7y8Lk0SiOwwrjKCnKPTCnzZOghT4JsuLHwSsqks0hhtKYipFlqtrsrMrqGt65qJJ4gaVNKt9Ruc8bXLSwoZqy0yQoWiLGoUBELxIyKOMKtaSoc2ixuICa3LsaoDLY87MNaw7gty3cvsu7iTt8vKbuBorbKGjbyqel75maTzArmkLrtg27iuhh6nJSybGi6WrpNBpauAxqH7MSx7tue3a1hqzKTL+zq6vC4mMIvO6scpnHKrxjppg+sHYM5imRqp3HXoWDLkaO-72Y4kXhs28Xecl-aQZ6+Xhcx0XlZ5naqr2aXZtl5mic1+SYRNpnzJJ62OtClnurRmFzfwo4TjOMy8FsWxEEmWxnleEBMBwPA6TgX4aHoRhmBuZ54yeIA). Maybe this is useful in some way ?

Reply:
thanks @icylace do you propose any change to the table ?

Reply:
> Would there be any use case for an app with no view & node ?

Yes for sure. Basically any time you want to use the state management features of hyperapp without rendering the state. The main case that comes to mind would be testing but perhaps also in an incremental adoption scenario.

Reply:
> thanks @icylace do you propose any change to the table ?

I have some thoughts. I'm thinking of making an edit.

Reply:
@kofifus @zaceno I added some clarifications to the props table.

Reply:
Thanks @icylace, I am a bit confused about Node required "Yes when view: is present"

It doesn't seem to produce any error if one ie only supply a node. 

If all values can be removed I think the column should be removed.

Reply:
@kofifus It's true that only providing `node:` is fine. However, if `view:` is provided while there is no `node:` then that creates a runtime error. Therefore, yes, `node:` is required to be present if `view:` is present.

Reply:
Ah ok got it!

OK so table is great, I think in the future the pages for view, node, subscriptons and Dispatch should include clear explanation of what happens if they are missing, but I don't think that should be in app.md/this PR

Are you otherwise happy for this to be merged ?

Reply:
Sure thing, I have no preference.